src/Microsoft.SqlTools.ServiceLayer/Agent/AgentService.cs

```diff
@@ -142,47 +141,27 @@ public void InitializeService(ServiceHost serviceHost)
-                    for (int i = 0; i < count; ++i)
-                    {
-                        var job = dt.Rows[i];
-                        if (JobUtilities.IsStep(job, sqlConnInfo))
-                        {
-                            var agentJobStep = JobUtilities.ConvertToAgentJobStep(job, sqlConnInfo);
-                            if (agentStepMap.ContainsKey(agentJobStep.RunDate))
-                            {
-                                agentStepMap[agentJobStep.RunDate].Add(agentJobStep);
-                            }
-                            else
-                            {
-                                var agentJobSteps = new List<AgentJobStep>();
-                                agentJobSteps.Add(agentJobStep);
-                                agentStepMap[agentJobStep.RunDate] = agentJobSteps;
-                            }
-                        }
-                        else
-                        {
-                            var agentJobHistoryInfo = JobUtilities.ConvertToAgentJobHistoryInfo(job, sqlConnInfo);
-                            agentJobs.Add(agentJobHistoryInfo);
-                        }
+                    List<AgentJobHistoryInfo> jobHistories = new List<AgentJobHistoryInfo>();
+                    if (count > 0) {
+                        var job = dt.Rows[0];
+                        string jobName = Convert.ToString(job[JobUtilities.UrnJobName], System.Globalization.CultureInfo.InvariantCulture);
+                        Guid jobId = (Guid) job[JobUtilities.UrnJobId];
+                        int runStatus = Convert.ToInt32(job[JobUtilities.UrnRunStatus], System.Globalization.CultureInfo.InvariantCulture);
+                        var t = new LogSourceJobHistory(jobName, sqlConnInfo, null, runStatus, jobId, null);
+                        var tlog = t as ILogSource;
+                        tlog.Initialize();
+                        var logEntries = t.LogEntries;
+                        jobHistories = JobUtilities.ConvertToAgentJobHistoryInfo(logEntries, job);
+                        tlog.CloseReader();
                     }
+                    result.Jobs = jobHistories.ToArray();
                     result.Succeeded = true;
-                    foreach (AgentJobHistoryInfo agentJobHistoryInfo in agentJobs)
-                    {
-                        if (agentStepMap.ContainsKey(agentJobHistoryInfo.RunDate))
-                        { 
-                            var agentStepList = agentStepMap[agentJobHistoryInfo.RunDate].ToList();
-                            agentStepList.Sort(delegate (AgentJobStep s1, AgentJobStep s2) { return s1.StepId.CompareTo(s2.StepId); });
-                            agentJobHistoryInfo.Steps = agentStepList.ToArray();
-                        }
-                    }
-                    result.Jobs = agentJobs.ToArray();
+                    ServerConnection connection = tuple.Item3;
```

Comment:
I'm not a huge fan of tuples, but if you're going to use them perhaps you should move all the values into name variables closer to where the tuple is created.  Basically moving this variable declaration to line 147.